targets/TARGET_Silicon_Labs/TARGET_EFM32/crc_api.c

```diff
@@ -0,0 +1,147 @@
+/***************************************************************************//**
+ * @file crc_api.c
+ *******************************************************************************
+ * @section License
+ * <b>(C) Copyright 2018 Silicon Labs, http://www.silabs.com</b>
+ *******************************************************************************
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+#include "device.h"
+#include "clocking.h"
+#include <stdio.h>
+
+#if DEVICE_CRC
+#include "mbed_assert.h"
+#include "crc_api.h"
+#include "em_cmu.h"
+
+#if defined(GPCRC_PRESENT) && (GPCRC_COUNT > 0)
+#include "em_gpcrc.h"
+
+static bool revOutput = false;
+static uint32_t final_xor;
+
+bool hal_crc_is_supported(const crc_mbed_config_t *config)
+{
+    //GPCRC supports any 16-bit poly, but only the CCITT 32-bit poly
```

Comment:
Looks good to me, just one query / concern, GPCRC_PRESENT will always be present for all targets? 
In case DEVICE_CRC is true and GPCRC_PRESENT is not defined `hal_crc_is_supported` should return false and rest of the API's should just return.

Reply:
DEVICE_CRC is a target-specific definition through targets.json, while GPCRC_PRESENT is a Silicon Labs CMSIS define. If a device doesn't have a GPCRC engine, then we don't add the CRC capability to its entry in targets.json.

The reason for still having the DEVICE_CRC define is to allow people to turn off CRC acceleration if they should want to do so.